src/Panel/Panel.php

```diff
@@ -89,14 +516,80 @@ public static function link(App $kirby): bool
         return true;
     }
 
+    /**
+     * Filters the data array and returns only
+     * entries requested via a Fiber partial
+     * request (or the whole array if full request).
+     *
+     * @param \Kirby\Cms\App $kirby
+     * @param string $component
+     * @param array $data
+     * @return array
+     */
+    public static function partial(App $kirby, string $component, array $data): array
+    {
+        // is it a partial request?
+        $request          = $kirby->request();
+        $include          = $request->header('X-Fiber-Include')   ?? get('_include');
+        $requestComponent = $request->header('X-Fiber-Component') ?? get('_component');
+
+        // split include string into an array of fields
+        $include = Str::split($include);
+
+        // if a full request is made, return all data
+        if (empty($include) === true) {
+            return $data;
+        }
+
+        // if a new component is requested, the include
+        // parameter must be ignored and all data returned
+        if (empty($requestComponent) === false && $requestComponent !== $component) {
+            return $data;
+        }
+
+        // otherwise filter data based on
+        // dot notation, e.g. `$props.tab.columns`
+        $partial = [];
+
+        // check if globals are requested and need to be merged
+        if (Str::contains(implode($include), '$')) {
```

Comment:
For better performance the original header/request value before splitting should be preserved in a variable. Then we don't need to implode it again here.